server/response_writer.go

```diff
@@ -0,0 +1,22 @@
+package server
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/ipni/storetheindex/api/v0/finder/model"
+	"github.com/multiformats/go-multihash"
+)
+
+type (
+	selectiveResponseWriter interface {
+		http.ResponseWriter
+		Accept(r *http.Request) error
+	}
+	lookupResponseWriter interface {
+		io.Closer
+		selectiveResponseWriter
+		Key() multihash.Multihash
+		WriteProviderResult(model.ProviderResult) error
+	}
+)
```

Comment:
OK to fork for now; eventually we should refactor the code copied from `caskadht` onto a reusable library.

Reply:
Agreed, I will take that one after the service is up. Too much of a boilerplate code for a service that is essentially a wrapper around  10 lines of code.